app/src/main/java/com/jerboa/MainActivity.kt

```diff
@@ -84,6 +84,7 @@ class JerboaApplication : Application() {
 
 class MainActivity : AppCompatActivity() {
     private val siteViewModel by viewModels<SiteViewModel>()
+    private val homeViewModel by viewModels<HomeViewModel>()
```

Comment:
I don't think we want to be doing this, as it is a partial reversion of #817. Is there another way we could approach this? Or failing that, we should make sure that we don't accidentally re-introduce issues that PR solved.

Reply:
@kensand 

Check out the community & community sidebar routes. They both have to share the same community view model which is similar to this case.

- Create a `navigation` with a routename (say Route.Graph.Home) and put both `BottomNavActivity` & `PostActivity` inside it.
- When creating the viewmodel use the parent backstack entry (ie Route.Graph.Home, see the community route on how to do that).
- Deeplinks will not work in nested navigation, so a redirect route has to be created after the `navigation` (again, see the redirect for community acitivity deeplink).

Reply:
Just realized the `PostActivity` can be opened from the community page as well. Should it then show only the posts from the community when swiping left and right?

Reply:
> Just realized the PostActivity can be opened from the community page as well. Should it then show only the posts from the community when swiping left and right?

Yes, it should. It looks like these navigation changes have added significantly complexity to this, especially considering that I now have to consider two different sources for posts, deeplinks, and a navigation system I have no experience with. I'm not sure how I want to proceed with this yet.

Reply:
Ok, I think I got navigation worked out - had to add a separate path & `PathActivity` for posts opened from a community, but it works as intended above. 

Made the changes as requested (I hope). Thanks for looking!